src/App/PropertyContainerPyImp.cpp

```diff
@@ -521,12 +521,20 @@ PyObject *PropertyContainerPy::getCustomAttributes(const char* attr) const
         getPropertyContainerPtr()->getPropertyMap(Map);
         PyObject *dict = PyDict_New();
         if (dict) {
-            for ( std::map<std::string,App::Property*>::iterator it = Map.begin(); it != Map.end(); ++it )
-#if PY_MAJOR_VERSION >= 3
-                PyDict_SetItem(dict, PyUnicode_FromString(it->first.c_str()), PyUnicode_FromString(""));
-#else
-                PyDict_SetItem(dict, PyString_FromString(it->first.c_str()), PyString_FromString(""));
-#endif
+            for ( std::map<std::string,App::Property*>::iterator it = Map.begin(); it != Map.end(); ++it )
+            {
+#if PY_MAJOR_VERSION >= 3
+                PyObject * key = PyUnicode_FromString(it->first.c_str());
```

Comment:
PyUnicode_FromString returns a new reference and PyDict_SetItem does not steal reference

Reply:
Should it be Py_XDECREF() instead, as PyUnicode_FromString() could in principle return NULL?  Otherwise, it may segfault when calling Py_DECREF() with nullptr.

Reply:
While you are correct in GENERAL, the property map should not have any nulls in it, AFAIK.

Reply:
A much better way is to use the PyCXX API here as it:
* automatically handles the ref counting
* raises an exception on errors
* simplifies the code a lot

Reply:
See 9f72a7b4061 and c331d608aa6